refactor: moved the TrackDetails Component to its own Container, and updated CD values to env vars

ihatelactose opened the pull request:
- Updated cd.yml to use env vars instead of using direct values
- Refactor:
   - Moved the TrackDetails Component into a separate container of its own
- Tests:
   - Wrote stubs and mock implementations for audio elements

Commit:
ci/cd: updated the cd.yml to use env vars

Commit:
ci/cd: updated the cd.yml, removed nodejs caching with yarn in production

tousifws commented:
Move `TrackDetails` to containers. It's connected to redux store. It should have its own saga, reducer, selector

Commit:
refactor: changed the TrackDetails component into a container

Commit:
fix: created stubs for called prototypes in tests
- Created stubs for HTMLAudioElement.prototype.pause/play for tests
- Created mock for router

tousifws commented on app/containers/TrackDetailsContainer/index.js:
```diff
@@ -68,20 +68,19 @@ export function TrackDetails({ dispatchGetDetails, details, error }) {
   useEffect(() => {
-    if (details.results) {
+    if (details.results || resultCount === 0) {
       setLoading(false);
```
Don't you think handling loading state from reducer would be better?

tousifws commented on app/containers/TrackDetailsContainer/index.js:
```diff
@@ -3,9 +3,9 @@
 import { createStructuredSelector } from 'reselect';
-import artistsContainerSaga from '@app/containers/ArtistsContainer/saga';
+import trackDetailsContainerSaga from '@app/containers/TrackDetailsContainer/saga';
 import { useRouter } from '@app/hooks/useRouter';
```
All the modules written by us should be under other libraries

tousifws commented on app/routeConfig.js:
```diff
@@ -2,15 +2,15 @@ import NotFound from '@containers/NotFoundPage/Loadable';
 import routeConstants from '@utils/routeConstants';
 import ArtistsContainer from '@containers/ArtistsContainer/index';
-import TrackDetails from './components/TrackDetails/index';
+import TrackDetailsContainer from './containers/TrackDetailsContainer/index';
```
Use relative import only if the file is in same directory otherwise use absolute import. Update them everywhere

tousifws commented:
Disable the annotation of jest coverage action. It makes it super annoying to review https://github.com/ArtiomTr/jest-coverage-report-action#change-annotations

Commit:
refactor: moved the loading state for contianer over to reducer
- updated the jest-coverage so as NOT to show the annotations over review
- Moved the loading state for TrackDetailsContainer over to reducer
- Updated selectors and wrote tests for the same
- Updated all the relative imports to absolute imports

Commit:
refactor: created loading in translations and moved the text over to intl

Commit:
update: updated TrackDetailsContainer with React.memo

Commit:
fix: fixed the TrackDetailsContainer reducer and component
- TrackDetailsContainer reducer now resets the details when requesting for a new track, preventing the cached track from showing up if the search fails
- TrackDetailsContainer main mounting component now properly renders if the searched track was found or not

tousifws reviewed (approved):
Do squash merge
